k8sinterface/k8sconfig.go

```diff
@@ -98,13 +97,18 @@ func GetK8sConfig() *restclient.Config {
 }
 
 // DEPRECATED
-func GetCurrentContext() *api.Context {
-	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{CurrentContext: clusterContextName})
-	config, err := kubeConfig.RawConfig()
-	if err != nil {
-		return nil
+func GetCurrentContext() *clientcmdapi.Context {
+	if kubeConfig := GetConfig(); kubeConfig != nil {
+		kubeConfig.CurrentContext = clusterContextName
+		if clusterContextName != "" {
+			if c, ok := kubeConfig.Contexts[clusterContextName]; ok {
+				return c
+			}
+		}
+		// if context name is not set, return the current context
+		return kubeConfig.Contexts[kubeConfig.CurrentContext]
```

Comment:
`kubeConfig.CurrentContext` will always be `clusterContextName` (due to assignment in line 102).. or am I missing something?

Reply:
You are right